Libraries/LibWeb/HTML/ElementInternals.cpp

```diff
@@ -46,6 +46,168 @@ GC::Ptr<DOM::ShadowRoot> ElementInternals::shadow_root() const
     return shadow;
 }
 
+// https://html.spec.whatwg.org/multipage/custom-elements.html#dom-elementinternals-setformvalue
+WebIDL::ExceptionOr<void> ElementInternals::set_form_value(Variant<GC::Root<FileAPI::File>, String, GC::Root<XHR::FormData>> value, Optional<Variant<GC::Root<FileAPI::File>, String, GC::Root<XHR::FormData>>> state)
+{
+    // 1. Let element be this's target element.
+    auto element = m_target_element;
+
+    // 2. If element is not a form-associated custom element, then throw a "NotSupportedError" DOMException.
+    if (!element->is_form_associated())
```

Comment:
The `is_form_associated()` method name is a bit confusing, since it's checking whether the element is a [form associated custom element](<https://html.spec.whatwg.org/multipage/custom-elements.html#form-associated-custom-element>), but the method name makes it seem like it's checking whether `element` is a `FormAssociatedElement`.

Reply:
Good point!